### DISCUSSION TITLE
Is there a way to control the order of server middle wares?

### DISCUSSION THREAD

**Discussion by chenyuncai** [nuxt/nuxt#17628] [Questions] [answered]
For now, the doc just tell us how to add a new middleware in the target folder and auto loaded by nuxt3, but how can i config the order to run?

**Comment by wobsoriano** [comment, marked as answer]
Did you try putting it instead in `serverMiddleware` property of nuxt config so maybe it can load from top to bottom?

**Reply by chenyuncai** [reply]
Thanks, that works!

**Reply by TJacoob** [reply]
@chenyuncai, could you provide a code example of how you achieved this?

So far I've only been able to run the middlewares in a reverse alphabetical order.

**Reply by wobsoriano** [reply]
> @chenyuncai, could you provide a code example of how you achieved this?
> 
> So far I've only been able to run the middlewares in a reverse alphabetical order.

Try something like

```typescript
export default defineNuxtConfig({
  serverMiddleware: [
    '~/server/middleware/first.ts',
    '~/server/middleware/second.ts',
    '~/server/middleware/third.ts'
  ]
})
```

**Reply by TJacoob** [reply]
@wobsoriano, thanks for the help, but I cannot get it to work. Right now, I tried to create three separate files to run it as you suggested it, and not only it does not follow the order, it actually runs every middleware, even if I don't add it to the `serverMiddleware` array.

The loading order as it seems to be is a reverse alphabetical. I name a file starting with z and it is run before all the others. Not quite sure if this is a bug or not, hence why I was curious with how @chenyuncai managed to make it work.

**Reply by chenyuncai** [reply]
Sorry, I havnt noticed this message. I have configured it in nuxt.config and it worked fine in 2021 when I just did a demo, But now it seems like the same problem as you said。
